contracts/lib/BlockDecoder.sol

```diff
@@ -169,7 +170,11 @@ library BlockDecoder {
                 require(vaddr == validators[i], "validator no exist");
                 idx = i;
             } else {
-                assert(false);
+                bool exist;
+                (exist, idx) = _validatorIndex(vaddr, validators);
```

Comment:
Need a way to make sure the idx is not used twice?  Perhaps a bitmap array may help.

Reply:
have solved this issue